src/components/editMeasure/populationCriteria/groups/QICore/QICoreMeasureGroups.tsx

```diff
@@ -307,21 +308,41 @@ const MeasureGroups = (props: MeasureGroupProps) => {
     );
   }, [measure?.elmJson]);
 
+  //default population displayId is: PopulationName_groupNumber
+  const getPopulationsWithDisplayId = (pops: Population[]): Population[] => {
+    return pops.map((population) => {
+      return {
+        ...population,
+        displayId: (
+          population.name.replace(" ", "") +
+          "_" +
+          (measureGroupNumber + 1)
+        ).replace(/^./, (match) => match.toUpperCase()),
+      };
+    });
+  };
+  const [allPopulationsWithDisplayId, setAllPopulationsWithDisplayId] =
+    useState<Population[]>(getPopulationsWithDisplayId(allPopulations));
+  useEffect(() => {
+    setAllPopulationsWithDisplayId(getPopulationsWithDisplayId(allPopulations));
+  }, [measureGroupNumber]);
+
```

Comment:
This should actually be in the useEffect underneath it. 

It already has measureGroupNumber in the dependency array.

Having two useEffects will trigger many rerenders.

Reply:
@mcmcphillips Thanks Matt! I moved it to the useEffect below.